token-metadata/program/src/state/mod.rs

```diff
@@ -156,3 +163,83 @@ where
     let pubkey_string = pubkey.as_ref().map(|p| p.to_string());
     serde::ser::Serialize::serialize(&pubkey_string, serializer)
 }
+
+/// Trait for resizable accounts.
+///
+/// Implementing this trait for a type will automatically allow the use of the `save` method,
+/// which can modify the size of an account.
+///
+/// A type implementing this trait must specify the `from_bytes` method, since an account can
+/// have variable size.
+pub trait Resizable: TokenMetadataAccount + BorshSerialize {
+    /// Saves the information to the specified account, resizing the account if needed.
+    ///
+    /// The account size can either increase or decrease depending on whether the account size
+    /// matches the struct size or not.
+    fn save<'a>(
+        &self,
+        account_info: &'a AccountInfo<'a>,
+        payer: &'a AccountInfo<'a>,
+        system_program: &'a AccountInfo<'a>,
+    ) -> Result<(), ProgramError> {
+        // the required account size
+        let required_size = Self::size();
+
+        if account_info.data_len() != required_size {
+            // no risk of overflow here since max account size is 10_000_000
+            let difference = required_size as i64 - account_info.data_len() as i64;
+            let snapshot = account_info.lamports();
+
+            if difference > 0 {
+                if difference as usize > MAX_PERMITTED_DATA_INCREASE {
+                    //return err!(CandyGuardError::DataIncrementLimitExceeded);
+                }
+
+                let lamports_diff = Rent::get()?
+                    .minimum_balance(required_size)
+                    .checked_sub(snapshot)
+                    .ok_or(MetadataError::AddressNotInReservation)?;
+
+                msg!("Funding {} lamports for account realloc", lamports_diff);
+
+                invoke(
+                    &system_instruction::transfer(payer.key, account_info.key, lamports_diff),
+                    &[payer.clone(), account_info.clone(), system_program.clone()],
+                )?;
+            } else {
+                let lamports_diff = snapshot
+                    .checked_sub(Rent::get()?.minimum_balance(required_size))
+                    .ok_or(MetadataError::AddressNotInReservation)?;
```

Comment:
This should be `MetadataError::NumericalOverflowError`.

Reply:
Fixed.